classes/command/class-remove-duplicate-postmeta.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * Remove duplicate postmeta command
+ *
+ * @package P4GBKS
+ */
+
+namespace P4GBKS\Command;
+
+/**
+ * Class for cleaning duplicate postmeta records.
+ */
+class Remove_Duplicate_Postmeta {
+	/**
+	 * Remove duplicate postmeta records
+	 *
+	 * @return boolean
+	 */
+	public function remove_duplicate() {
+
+		global $wpdb;
+
+		// phpcs:disable
+		$sql = "DELETE t1 FROM wp_postmeta t1 
```

Comment:
There are some postmeta keys that actually need to have multiple values. I don't know if there are many of these, but I guess it would be prudent to work with a whitelist here of keys that we know can be removed?

Reply:
Maybe you can then change the semantics a bit, so the classname is `Duplicated_Postmeta`. Then you could name the function `Duplicated_Postmeta::remove()`. You could then add the logic that is now in the admin page controller for detection (https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/271/files#diff-ccb958aec6798a7ea30f1315a3632727R89) in a function called `Duplicated_Postmeta::detect()`.

Reply:
> There are some postmeta keys that actually need to have multiple values. I don't know if there are many of these, but I guess it would be prudent to work with a whitelist here of keys that we know can be removed?

I was unaware need of postmeta key with multiple values, could you please provide some examples?

Reply:
> Maybe you can then change the semantics a bit, so the classname is `Duplicated_Postmeta`. Then you could name the function `Duplicated_Postmeta::remove()`. You could then add the logic that is now in the admin page controller for detection (https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/271/files#diff-ccb958aec6798a7ea30f1315a3632727R89) in a function called `Duplicated_Postmeta::detect()`.

- yes It will be more suitable. I will update it.

Reply:
> > There are some postmeta keys that actually need to have multiple values. I don't know if there are many of these, but I guess it would be prudent to work with a whitelist here of keys that we know can be removed?
> 
> I was unaware need of postmeta key with multiple values, could you please provide some examples?

It looks like wpml depends on this, they call `get_post_meta` with the `$single` parameter` to false and then loop over it in here:
persistence/app/public/wp-content/plugins/sitepress-multilingual-cms/inc/wpml-api.php:625

Reply:
> There are some postmeta keys that actually need to have multiple values. I don't know if there are many of these, but I guess it would be prudent to work with a whitelist here of keys that we know can be removed?

I added a meta_key whitelist at [here](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/271/files#diff-6909d1fef617f885c6b17e50fc90b6f4R20)